src/math.rs

```diff
@@ -260,10 +267,27 @@ impl<S: BaseFloat> From<&Aabb3<S>> for Obb<S> {
 
 impl<S: BaseFloat> From<Aabb3<S>> for Obb<S> {
     fn from(aabb: Aabb3<S>) -> Self {
-        Obb::new(
-            Isometry3::new(Quaternion::one(), EuclideanSpace::to_vec(aabb.center())),
-            aabb.dim() / (S::one() + S::one()),
-        )
+        Self::from(&aabb)
+    }
+}
+
+impl<S: BaseFloat> From<&OrientedBeam<S>> for Obb<S> {
+    fn from(beam: &OrientedBeam<S>) -> Self {
+        let beam_isometry = beam.isometry_inv.inverse();
+        let earth_radius_mean = S::from(0.5 * (EARTH_RADIUS_MIN_M + EARTH_RADIUS_MAX_M)).unwrap();
+        let z_off = earth_radius_mean - beam_isometry.translation.magnitude();
+        let pt_off = Point3::new(S::zero(), S::zero(), z_off);
+        let translation = beam_isometry.rotation.rotate_point(pt_off) + beam_isometry.translation;
```

Comment:
Nit: If you use a Vector3 in the line above, that's just multiplication with the beam isometry